homeassistant/components/proxmoxve/sensor.py

```diff
@@ -0,0 +1,194 @@
+                                virtual_machine,
+                                sensor,
+                            )
+                        )
+
+            for container in node[CONF_CONTAINERS]:
+                for sensor in SENSOR_TYPES:
+                    if GUESTTYPE_LXC in SENSOR_TYPES[sensor][4]:
+                        sensors.append(
+                            ProxmoxSensor(
+                                hass.data[PROXMOX_CLIENTS][
+                                    f"{entry[CONF_HOST]}:{port}"
+                                ],
+                                node["node"],
+                                GUESTTYPE_LXC,
+                                container,
+                                sensor,
+                            )
+                        )
+
+    async_add_devices(sensors, True)
+
+
+class ProxmoxSensor(Entity):
+    """A sensor for reading Proxmox VE data."""
+
+    def __init__(self, proxmox_client, item_node, item_type, item_id, sensor_type):
+        """Initialize the sensor."""
+        self._available = True
+        self._device_class = SENSOR_TYPES[sensor_type][3]
+        self._icon = SENSOR_TYPES[sensor_type][2]
+        self._item_node = item_node
+        self._item_type = item_type
+        self._item_id = item_id
+        self._name = None
+        self._proxmox_client = proxmox_client
+        self._sensor_type = sensor_type
+        self._state = None
+        self._unit_of_measurement = SENSOR_TYPES[sensor_type][1]
+
+    @property
+    def name(self):
+        """Return the name of the entity."""
+        return self._name
+
+    @property
+    def device_class(self):
+        """Return the class of this sensor."""
+        return self._device_class
+
+    @property
+    def icon(self):
+        """Icon to use in the frontend, if any."""
+        return self._icon
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+        return self._state
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement of this entity, if any."""
+        return self._unit_of_measurement
+
+    @property
+    def available(self):
+        """Return True if entity is available."""
+        return self._available
+
+    def update(self):
+        """Update sensore values."""
+        items = (
+            self._proxmox_client.get_api_client()
+            .nodes(self._item_node)
+            .get(self._item_type)
+        )
+        item = next(
+            (item for item in items if item["vmid"] == str(self._item_id)), None
+        )
+
+        if item is None:
+            _LOGGER.warning("Couldn't find VM/Container with the ID %s", self._item_id)
+            return None
+
+        self._name = (
+            f"{self._item_node} {item['name']} {SENSOR_TYPES[self._sensor_type][0]}"
+        )
+
+        item["disk"] = int(item["disk"])
+        item["maxdisk"] = int(item["maxdisk"])
+
+        if self._sensor_type == "cpu":
```

Comment:
could these be converted to lambda functions put into the SENSOR_TYPES array? This would reduce all these elifs to one call of the lambda for whichever type it is.

Reply:
HI @jhollowe , thanks for your comment, but unfortunately I´m not familiar with lambda functions, that why may I ask you to give me an concrete example?

Reply:
so in SENSOR_TYPES you might have `cpu_use_percent": ["CPU use", PERCENTAGE, "mdi:chip", None, SENSOR_AVAIL_ALL, lamdba itm : round(itm["cpu"] * 100, 1)],`

and then you would pass that and store that in the sensor (let's call it "_getter"). You can then replace all these elifs with `self._state = self._getter(item)`.
Basically, you store the function needed to get the value for each type of sensor in the sensor itself.

Reply:
Thanks for that great idea 😃